Add support for URN and VIAF

wgresshoff opened the pull request:
:heart: Thank you for your contribution!

### Description

To add an URN service API wrapper (offered by DNB) analogue to Datacite service wrapper a function for URN normalizaion is needed. VIAF support is not really needed for this but to better support imports of miami user data.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [ ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

lnielsen commented on idutils/__init__.py:
```diff
@@ -661,6 +669,11 @@ def is_ror(val):
     return ror_regexp.match(val)
 
 
+def is_viaf(val):
```
Do you mind adding some tests in https://github.com/inveniosoftware/idutils/blob/master/tests/test_idutils.py#L21

```python
(
        "http://viaf.org/viaf/1234",  # test value
        ["viaf", ], # type identification
        "1234",   # normalised value
        "http://viaf.org/viaf/1234",  # landing page value
    ),
```


There's one thing I'm slightly concerned about is what happens in the ``test_idempotence``:

```
val = "http://viaf.org/viaf/1234"
new_val = normalise_pid(val, "viaf")
assert detect_identifier_schemes(new_val, "viaf") == ["viaf"]
```

Basically if you normalise a viaf, and run a detection on the normalised value, will it still be identified as a viaf. Because it's just an integer, I think pubmed will kick in instead. I think we had a similar issue with GND

wgresshoff replied:
Ah, that is the reason a gnd is normalized to `gnd:xyz`? I will add some tests and look how to make it safe. Thanks!

wgresshoff commented:
I have changed the viaf recognition based on a regexp I found in wikidata, and the viaf normalisation to "viaf:..." as the gnd does. I also added a test and to make it run had to change the id type detection, too.
